src/ClimaAtmos/Dycore/src/Grids.jl

```diff
@@ -207,4 +207,134 @@ function computegeometry(topology::AbstractTopology{dim}, D, ξ, ω, meshwarp,
 end
 # }}}
 
+
+# {{{ compute element size for box domain 
+function compute_element_size(dim, Nq, vgeo, e) 
+
+    DFloat = eltype(vgeo)
+    
+    if (dim == 2)
+        
+        x, y = zeros(DFloat, 4), zeros(DFloat, 4)
+        
+        x[1], y[1] = vgeo[1, 1,   _x, e], vgeo[1, 1,   _y, e]
+        x[2], y[2] = vgeo[Nq, 1,  _x, e], vgeo[Nq, 1,  _y, e]
+        x[3], y[3] = vgeo[1, Nq,  _x, e], vgeo[1, Nq,  _y, e]
+        x[4], y[4] = vgeo[Nq, Nq, _x, e], vgeo[Nq, Nq, _y, e]        
+
+        #Element sizes (as if it were linear)
+        dx = maximum(x[:]) - minimum(x[:])
+        dy = maximum(y[:]) - minimum(y[:])
+
+        
+        #Average distance between LGL points inside the element:
+        dx_mean = dx/max(Nq - 1, 1)
+        dy_mean = dy/max(Nq - 1, 1)
+        
+        ds = (dx, dy, dx_mean, dy_mean)
+    
+    elseif (dim == 3)
+        
+        x, y, z = zeros(DFloat, 8), zeros(DFloat, 8), zeros(DFloat, 8)
+         
+        x[1], y[1], z[1] = vgeo[1, 1,   1, _x, e], vgeo[1, 1,   1, _y, e], vgeo[1, 1,   1, _z, e]
```

Comment:
This assumes that each processor has an element and that all the elements on the processor are the same size. Both of which might not be true.